todo/src/components/TodoList.js

```diff
@@ -1,34 +1,105 @@
 import React, { useState } from "react"
 import TodoItem from "./TodoItem"
 import uuid from "react-uuid"
+import { DragDropContext, Droppable, Draggable } from "react-beautiful-dnd"
+import "../App.css"
 
 function TodoList(props) {
 	const [task, setTask] = useState("Add a task")
 	const [todos, setTodos] = useState(props.todos)
+	const [counter, setCounter] = useState(0)
+	const [filter, setFilter] = useState(" All")
+
+	const onDragEnd = (result) => {
+		const { destination, source } = result
+
+		if (!destination) {
+			return
+		}
+
+		if (destination.index === source.index) {
+			return
+		}
+
+		const newTasks = Array.from(todos)
+		const [removed] = newTasks.splice(source.index, 1)
+		newTasks.splice(destination.index, 0, removed)
+		setTodos(newTasks)
+	}
 
 	const handleChange = (event) => {
 		setTask(event.target.value)
 	}
 
 	const handleSubmit = (event) => {
 		event.preventDefault()
-		const newTodo = { content: task, isComplete: false }
-		setTodos([...todos, newTodo])
-		setTask("Add a task")
+		if (!(task === "Add a task" || task === "")) {
+			const newTodo = { content: task, isComplete: false, id: uuid() }
+			setTodos([...todos, newTodo])
+			setCounter(counter + 1)
+		}
+		setTask("")
 	}
+	const deleteTask = (id) => {

```

Comment:
`deleteTask`, `clearCompleted`, and `checkBoxChange` are declared here but are used only in another component. It creates unnecessary complexity.

Reply:
fixed